amulet/charm.py

```diff
@@ -1,31 +1,64 @@
+from .helpers import reify
+from .helpers import run_bzr
+from .helpers import setup_bzr
+from charmworldlib.charm import Charm
+from path import path
+from path import tempdir
 import os
-import yaml
+import shlex
 import shutil
+import subprocess
 import tempfile
+import yaml
 
-from charmworldlib.charm import Charm
-from .helpers import run_bzr, setup_bzr
 
+class CharmCache(dict):
+    def __init__(self, test_charm):
+        super(CharmCache, self).__init__()
+        self.test_charm = test_charm
+
+    @staticmethod
+    def get_charm(charm_path, branch=None, series='precise'):
+        if charm_path.startswith('lp:'):
+            return LaunchpadCharm(charm_path)
+        elif branch and branch.startswith('lp:'):
+            return LaunchpadCharm(branch)
+
+        if charm_path.startswith('local:'):
+            return LocalCharm(
+                os.path.join(
+                    os.environ.get('JUJU_REPOSITORY', ''),
+                    charm_path[len('local:'):]))
+
+        if branch and branch.endswith('.git'):
```

Comment:
It would be nice if it also supported a gh: leader that defaulted to GitHub, like lp: defaults to Launchpad.  I believe bundletester already uses this convention.